apps/api/src/app/inbox/usecases/session/session.usecase.ts

```diff
@@ -161,4 +210,421 @@ export class Session {
+      // Ensure we have at least 4 bytes (8 hex chars)
+      return true; // Invalid timestamp format, consider expired
+    }
+
+    try {
+      const createdDateTimestamp = timestampHexToDate(createdDate);
+      const now = new Date();
+      const diffTimeInHours = differenceInHours(now, createdDateTimestamp);
+
+      if (diffTimeInHours > KEYLESS_RETENTION_TIME_IN_HOURS) {
+        return true;
+      }
+    } catch (error) {
+      this.logger.error({ err: error }, 'Error parsing timestamp');
+
+      // If there's any error parsing the timestamp, consider it expired
+      return true;
+    }
+
+    return false;
+  }
+
+  async processKeyless(): Promise<EnvironmentResponseDto> {
+    if (process.env.NOVU_ENTERPRISE !== 'true') {
+      throw new BadRequestException('Keyless is not supported in community edition');
+    }
+
+    const organization = await this.communityOrganizationRepository.findById(process.env.KEYLESS_ORGANIZATION_ID!);
+
+    if (!organization) {
+      this.logger.error('Keyless Organization not found');
+      throw new InternalServerErrorException('Keyless Organization not found');
+    }
+
+    const user = await this.communityUserRepository.findByEmail(process.env.KEYLESS_USER_EMAIL!);
+
+    if (!user) {
+      throw new InternalServerErrorException('Keyless User not found');
+    }
+
+    const key = `sk_${await this.generateUniqueApiKey.execute()}`;
+    const encryptedApiKey = encryptApiKey(key);
+    const hashedApiKey = createHash('sha256').update(key).digest('hex');
+
+    const encodedDate = dateToTimestampHex(new Date());
+    const identifier = `${this.KEYLESS_ENVIRONMENT_PREFIX}${encodedDate}_${shortId(4)}`;
+    const environment = await this.environmentRepository.create({
+      _organizationId: organization._id,
+      name: `Keyless ${new Date().toISOString()}`,
+      identifier,
+      apiKeys: [
+        {
+          key: encryptedApiKey,
+          _userId: user._id,
+          hash: hashedApiKey,
+        },
+      ],
+    });
+
+    await this.createNovuIntegrationsUsecase.execute(
+      CreateNovuIntegrationsCommand.create({
+        environmentId: environment._id,
+        organizationId: environment._organizationId,
+        userId: user._id,
+        name: 'Keyless Integration',
+        channels: [ChannelTypeEnum.IN_APP],
+      })
+    );
+
+    await this.createWorkflowsUsecase(environment._id, environment._organizationId, user._id);
+
+    return this.convertEnvironmentEntityToDto(environment);
+  }
+
+  async createWorkflowsUsecase(environmentId: string, organizationId: string, userId: string) {
+    const inAppTemplate = await this.messageTemplateRepository.create({
+      type: StepTypeEnum.IN_APP,
+      content: '',
+      avatar: 'https://dashboard-v2.novu.co/images/info.svg',
+      _environmentId: environmentId,
+      _organizationId: organizationId,
+      _creatorId: userId,
+      active: true,
+      name: 'In-App Notification',
+      controls: {
```

Comment:
@djabarovgeorge I have told you, not a huge fan of creating it directly on the DAL, would be so much cleaner to use the workflow usecase. Not critical tho